aws-iot-softwarepackage/src/main/java/software/amazon/iot/softwarepackage/CreateHandler.java

```diff
@@ -0,0 +1,188 @@
+package software.amazon.iot.softwarepackage;
+
+import com.amazonaws.util.StringUtils;
+import org.apache.commons.collections.MapUtils;
+import software.amazon.awssdk.services.iot.IotClient;
+import software.amazon.awssdk.services.iot.model.CreatePackageRequest;
+import software.amazon.awssdk.services.iot.model.CreatePackageResponse;
+import software.amazon.awssdk.services.iot.model.CreatePackageVersionRequest;
+import software.amazon.awssdk.services.iot.model.CreatePackageVersionResponse;
+import software.amazon.awssdk.services.iot.model.IotException;
+import software.amazon.awssdk.services.iot.model.UpdatePackageRequest;
+import software.amazon.awssdk.services.iot.model.UpdatePackageResponse;
+import software.amazon.awssdk.services.iot.model.UpdatePackageVersionRequest;
+import software.amazon.awssdk.services.iot.model.UpdatePackageVersionResponse;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+public class CreateHandler extends BaseHandlerStd {
+    private final static String OPERATION = "CreateSoftwarePackage";
+    private static final String CALL_GRAPH = "AWS-IoT-SoftwarePackage::Create";
+    private static final int MAX_PACKAGE_NAME_LENGTH = 128;
+    private Logger logger;
+    private String clientToken;
+
+    @Override
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<IotClient> proxyClient,
+            final Logger logger) {
+
+        this.logger = logger;
+        this.clientToken = request.getClientRequestToken();
+
+        final ResourceModel resourceModel = request.getDesiredResourceState();
+
+        // create a package name if not provided by user
+        if (StringUtils.isNullOrEmpty(resourceModel.getPackageName())) {
+            resourceModel.setPackageName(generateName(request));
+        }
+
+        resourceModel.setDefaultVersionName(DEFAULT_PACKAGE_VERSION_NAME);
```

Comment:
This will override the defaultVersionName if it's already set by user

Reply:
removed default version name as part of the schema following conversations with the team.